packages/mui-lab/src/Masonry/Masonry.js

```diff
@@ -182,80 +182,99 @@ const Masonry = React.forwardRef(function Masonry(inProps, ref) {
 
   const classes = useUtilityClasses(ownerState);
 
-  React.useEffect(() => {
-    const handleResize = () => {
-      const parentWidth = masonryRef.current.clientWidth;
-      const childWidth = masonryRef.current.firstChild.clientWidth;
-      const firstChildComputedStyle = window.getComputedStyle(masonryRef.current.firstChild);
-      const firstChildMarginLeft = parseToNumber(firstChildComputedStyle.marginLeft);
-      const firstChildMarginRight = parseToNumber(firstChildComputedStyle.marginRight);
-
-      if (parentWidth === 0 || childWidth === 0) {
-        return;
-      }
+  const handleResize = (elements) => {
```

Comment:
Could we please move it outside of the component's function

Reply:
This is not resolved, move the method outside of the component

Reply:
`setMaxColumnHeight` or `setNumberOfLineBreaks` should be used, so we should keep this inside :)